src/controller/python/chip/clusters/Command.py

```diff
@@ -296,6 +298,32 @@ def SendCommand(future: Future, eventLoop, responseType: Type, device, commandPa
         ))
 
 
+def _BuildPyInvokeRequestData(commands: List[InvokeRequestInfo], timedRequestTimeoutMs: Optional[int], responseTypes, suppressTimedRequestMessage: bool = False) -> List[PyInvokeRequestData]:
+    numberOfCommands = len(commands)
+    pyBatchCommandsDataArrayType = PyInvokeRequestData * numberOfCommands
+    pyBatchCommandsData = pyBatchCommandsDataArrayType()
+    for idx, command in enumerate(commands):
+        clusterCommand = command.Command
+        responseType = command.ResponseType
+        if (responseType is not None) and (not issubclass(responseType, ClusterCommand)):
+            raise ValueError("responseType must be a ClusterCommand or None")
+        if clusterCommand.must_use_timed_invoke and timedRequestTimeoutMs is None or timedRequestTimeoutMs == 0:
+            if not suppressTimedRequestMessage:
+                raise chip.interaction_model.InteractionModelError(chip.interaction_model.Status.NeedsTimedInteraction)
+
+        payloadTLV = clusterCommand.ToTLV()
+
+        pyBatchCommandsData[idx].commandPath.endpointId = c_uint16(command.EndpointId)
+        pyBatchCommandsData[idx].commandPath.clusterId = c_uint32(clusterCommand.cluster_id)
+        pyBatchCommandsData[idx].commandPath.commandId = c_uint32(clusterCommand.command_id)
+        pyBatchCommandsData[idx].tlvData = cast(c_char_p(bytes(payloadTLV)), c_void_p)
+        pyBatchCommandsData[idx].tlvLength = c_size_t(len(payloadTLV))
+
+        responseTypes.append(responseType)
+
+    return pyBatchCommandsData
+
+
 def SendBatchCommands(future: Future, eventLoop, device, commands: List[InvokeRequestInfo],
                       timedRequestTimeoutMs: Optional[int] = None, interactionTimeoutMs: Optional[int] = None, busyWaitMs: Optional[int] = None,
                       suppressResponse: Optional[bool] = None) -> PyChipError:
```

Comment:
nit: maybe indenting should be a bit different as `None` seems to be another variant of `returns the cluster-object equivalent ...`? 

The signature says it returns PyChipError ... so does it return the error code or does it raise an exception? The text and signature seem to conflict a bit here.